system/View/View.php

```diff
@@ -402,35 +402,41 @@ public function extend(string $layout)
 	/**
 	 * Starts holds content for a section within the layout.
 	 *
-	 * @param string $name
+	 * @param string $name Section name
+	 *
+	 * @return void
 	 */
-	public function section(string $name)
+	public function section(string $name): void
 	{
-		$this->currentSection = $name;
+		$this->sectionStack[] = $name;
```

Comment:
I know it may sound absurd but maybe someone is using the value of `$this->currentSection` after this method is called. I'm thinking we leave it as-is for preserving BC for now then we remove this line in the future with a `@todo` comment.

Reply:
Hope I implemented your suggestion correctly.